Work on #33 (multi-line improvements) and #36 (variables)

JBartelsson opened the pull request:
multiple Lines are now in.

> "--"
> is the trigger Command for a comment!

JBartelsson committed:
Update FileParser.js

GnaspGames commented:
Note: This is for #33

JBartelsson committed:
Vars are now in

JBartelsson committed:
Variables work differently now

Define a variable with $aVAR=aVALUE
and use it like
/say aVAR
You can use already assigned VARS in the assignments of new VARS
$aVAR2=aVAR

GnaspGames commented on FileParser.js:
```diff
@@ -253,7 +254,24 @@ var FileParser = (function ()
+		// //there is a bug with the re-assignment of variables here. I couldn't find it!
+		// var VarValue = (line.substring(line.indexOf("=")+1)).trim();
+    	var parts = line.split('=', 2);
+   	 	var varName = parts[0].replace("$","");
```
I thought you decided not to replace the `$` character? You'll have to add it back in here if that's the case:

```
line = line.replace(new RegExp("\\"+Vars,'g'), this.CustomVariables[Vars]);
```

GnaspGames commented:
## Copying from conversation for future reference:

Regarding this:

```
if (line[0]+line[1]!="--"){
            for (var Vars in this.CustomVariables){
            line = line.replace(new RegExp("\\"+Vars,'g'), this.CustomVariables[Vars]);
        }
            this.FinalCommand += " " + line;
        }
```

Is it is, that is really messy at the moment.

Put this into it's own function:

```
            for (var Vars in this.CustomVariables){
            line = line.replace(new RegExp("\\"+Vars,'g'), this.CustomVariables[Vars]);
        }
```

Then call it at the beginning of `processRowLine`, `processJsonLine`,`processCommandBlockLine` and `processBangLine`.

Then inside `processVariableLine` only use it to parse `varValue`

GnaspGames commented:
Note: This will also be going towards #36 - variables

JBartelsson committed:
Variables should work now!

Example:
#Start
>{"type":"impulse","auto":true}
$Name=Johann
/say $Name
>{"type":"chain","auto":true}
$message=Hello $Name
/say $message
$message=Goodbye $Name
/say $message
